docs/LedgerEvents.md

```diff
@@ -0,0 +1,124 @@
+MIR certificates.
+The field `mir` is a record which contains:
+* A map from stake credentials to lovelace, to be taken from the reserves.
+* A map from stake credentials to lovelace, to be taken from the treasury.
+* Additions to the reserves, being taken from the treasury.
+* Additions to the treasury, being taken from the reserves.
+
+### `NoMirTransfer mir availableReserves availableTreasury`
+
+This event only occurs on epoch boundaries,
+it describes a failed MIR certificate (due to a lack of funds in the reserves or treasury).
+The field `mir` is the same as in the `MirTransfer` event.
+The fields `availableReserves` and `availableTreasury` give the total available lovelace
+that a MIR certificate can utilize from the reserves or the treasury (respectively).
+
+### `RestrainedRewards e eraIgnored unregistered`
+
+This event only occurs on epoch boundaries,
+it describes rewards which are calculated by the ledger, but are not
+delivered to a stake address.
+The field `e` is the epoch number of the new epoch.
+The field `eraIgnored` corresponds to the rewards not delivered
+due to the Shelley bug described in Section 17.4 of the Shelley ledger specification.
+The field `unregistered` corresponds to the rewards not delivered
+due to the account being de-registered.
+
+### `RupdEvent e rewards`
+
+This event describes the new reward calculations done as a part of the
+incremental reward calculation.
+The field `e` is the epoch when these rewards become available
+(the epoch following the one when the calculation is performed).
+Some of these rewards may not end up being delivered, due to the reasons
+described in the `RestrainedRewards` event.
+The field `rewards` is a map from stake credentials to lovelace.
+
+### `TotalRewardEvent e regRU`
+
+This event only occurs on epoch boundaries, it contains all the rewards which
+are delivered on the epoch boundary.
+The field `e` is the epoch number of the new epoch.
+It is related to the other rewards events by the property:
+```
+RupdEvent - RestrainedRewards = Total
+```
+
+### `TotalDeposits totalDeposits`
+
+This event happens for every transaction, it contains the sum of all the deposits
+paid by the given transaction (both stake credential registration deposits and
+stake pool registration deposits).
+
+### `RegisterPool poolID`
+This event happens for every new stake pool registration certificate.
+The field `poolID` is the stake pool ID.
+
+### `ReregisterPool hk`
+This event happens for every stake pool re-registration certificate
+(a change in stake pool parameters).
+The field `poolID` is the stake pool ID.
+
+### `SuccessfulPlutusScriptsEvent debugInformation`
+This event happens for every collection of Plutus scripts in a given transaction
+which successfully execute.
+The field `debugInformation` is a list of `PlutusDebug`, which is a record containing
+all the information needed to re-run a Plutus script.
+
+### `FailedPlutusScriptsEvent (scriptFailuresToPlutusDebug fs`
```

Comment:
```suggestion
### `FailedPlutusScriptsEvent (scriptFailuresToPlutusDebug fs)`
```

Reply:
oops, that's a copy paste error, should have been `FailedPlutusScriptsEvent debugInformation` to match the event above. good catch!